From ccf5332c9a11ddd0242daa965306cae0710a1bb6 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 12 Jan 2014 21:32:19 +0100
Subject: Look for the datafile next to the executable.

XRick can be run from Deskbar or Tracker this way.

diff --git a/src/xrick.c b/src/xrick.c
index 38d4e26..870adc7 100644
--- a/src/xrick.c
+++ b/src/xrick.c
@@ -25,8 +25,13 @@ main(int argc, char *argv[])
 	sys_init(argc, argv);
 	if (sysarg_args_data)
 		data_setpath(sysarg_args_data);
-	else
-		data_setpath("data.zip");
+	else {
+		const char* basename = dirname(argv[0]);
+		char* datapath = malloc(strlen(basename) + strlen("/data.zip") + 1);
+		strcpy(datapath, basename);
+		strcat(datapath, "/data.zip");
+		data_setpath(datapath);
+	}
 	game_run();
 	data_closepath();
 	sys_shutdown();
-- 
2.37.3


From fb1d68079b8cf99c7c99fb775d1a33ec2edf8ad7 Mon Sep 17 00:00:00 2001
From: Humdinger <humdingerb@gmail.com>
Date: Mon, 26 May 2014 19:18:37 +0200
Subject: Disable quitting with ESC, just end the currently running game.

Change default control keys to avoid clashes with most keymaps.

diff --git a/src/sysevt.c b/src/sysevt.c
index 9e3830f..ea478c7 100644
--- a/src/sysevt.c
+++ b/src/sysevt.c
@@ -71,11 +71,11 @@ processEvent()
       SETBIT(control_status, CONTROL_END);
       control_last = CONTROL_END;
     }
-    else if (key == syskbd_xtra) {
+/*  else if (key == syskbd_xtra) {
       SETBIT(control_status, CONTROL_EXIT);
       control_last = CONTROL_EXIT;
     }
-    else if (key == syskbd_fire) {
+*/    else if (key == syskbd_fire) {
       SETBIT(control_status, CONTROL_FIRE);
       control_last = CONTROL_FIRE;
     }
@@ -137,11 +137,11 @@ processEvent()
       CLRBIT(control_status, CONTROL_END);
       control_last = CONTROL_END;
     }
-    else if (key == syskbd_xtra) {
+/*  else if (key == syskbd_xtra) {
       CLRBIT(control_status, CONTROL_EXIT);
       control_last = CONTROL_EXIT;
     }
-    else if (key == syskbd_fire) {
+*/    else if (key == syskbd_fire) {
       CLRBIT(control_status, CONTROL_FIRE);
       control_last = CONTROL_FIRE;
     }
diff --git a/src/syskbd.c b/src/syskbd.c
index 2394760..adde88f 100644
--- a/src/syskbd.c
+++ b/src/syskbd.c
@@ -17,11 +17,11 @@
 
 U8 syskbd_up = SDLK_o;
 U8 syskbd_down = SDLK_k;
-U8 syskbd_left = SDLK_z;
-U8 syskbd_right = SDLK_x;
+U8 syskbd_left = SDLK_a;
+U8 syskbd_right = SDLK_s;
 U8 syskbd_pause = SDLK_p;
-U8 syskbd_end = SDLK_e;
-U8 syskbd_xtra = SDLK_ESCAPE;
+U8 syskbd_end = SDLK_ESCAPE;
+U8 syskbd_xtra = SDLK_F12;
 U8 syskbd_fire = SDLK_SPACE;
 
 /* eof */
-- 
2.37.3


From 70b7bf437ab92f2a71c48b5046b09122ae145dae Mon Sep 17 00:00:00 2001
From: Begasus <begasus@gmail.com>
Date: Thu, 29 Jun 2023 13:01:09 +0000
Subject: Makefile patches


diff --git a/Makefile b/Makefile
index d87bc0e..eddc21c 100644
--- a/Makefile
+++ b/Makefile
@@ -71,10 +71,8 @@ endif
 all:
 	@echo "ROOTDIR=$(ROOTDIR)" > Makefile.global
 	@echo "XOBJ=$(XOBJ)" >> Makefile.global
-	@echo "CFLAGS=-g -ansi -pedantic -Wall -W -O2 -I $(ROOTDIR)/include $(shell sdl-config --cflags)" >> Makefile.global
-	@echo "LDFLAGS=-lz $(shell sdl-config --libs)" >> Makefile.global
-	@echo "CC=gcc" >> Makefile.global
-	@echo "CPP=gcc -E" >> Makefile.global
+	@echo "CFLAGS+=-pedantic -Wall -I$(ROOTDIR)/include $(shell sdl-config --cflags)" >> Makefile.global
+	@echo "LIBS+=-lz $(shell sdl-config --libs)" >> Makefile.global
 	$(MAKE) -C src all
 
 clean:
diff --git a/src/Makefile b/src/Makefile
index fb1c7eb..147f143 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -40,10 +40,10 @@ OBJECTS = unzip.o data.o scr_xrick.o scr_pause.o scr_imain.o scr_imap.o scr_game
 all: $(TARGET)
 
 $(TARGET): $(OBJECTS) $(XOBJ)
-	$(CC) $(OBJECTS) $(XOBJ) $(LDFLAGS) -o $(TARGET)
+	$(CC) $(LDFLAGS) $(OBJECTS) $(XOBJ) -o $(TARGET) $(LIBS)
 
 %.o: %.c
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 xrick.res: xrick.rc
 	windres $< -O coff -o $@
-- 
2.37.3


From f0965542e16acc9a77b8f55ac17bcdb7b1c13d9d Mon Sep 17 00:00:00 2001
From: Begasus <begasus@gmail.com>
Date: Thu, 29 Jun 2023 13:08:02 +0000
Subject: Fix multiple defenitions


diff --git a/include/img.h b/include/img.h
index dbd5798..e4808cb 100644
--- a/include/img.h
+++ b/include/img.h
@@ -27,7 +27,7 @@ typedef struct {
   U8 *pixels;
 } img_t;
 
-img_t *IMG_SPLASH;
+extern img_t *IMG_SPLASH;
 
 #endif
 
diff --git a/src/dat_snd.c b/src/dat_snd.c
index d28705d..42343c4 100644
--- a/src/dat_snd.c
+++ b/src/dat_snd.c
@@ -19,14 +19,14 @@
 
 sound_t *WAV_WAA;
 sound_t *WAV_BOMB;
-sound_t *WAV_BULLET;
-sound_t *WAV_WALK;
-sound_t *WAV_JUMP;
+extern sound_t *WAV_BULLET;
+extern sound_t *WAV_WALK;
+extern sound_t *WAV_JUMP;
 sound_t *WAV_TING;
-sound_t *WAV_BOMBSHHT;
-sound_t *WAV_BONUS;
+extern sound_t *WAV_BOMBSHHT;
+extern sound_t *WAV_BONUS;
 sound_t *WAV_SHHT;
-sound_t *WAV_BOX;
+extern sound_t *WAV_BOX;
 sound_t *WAV_DDDING;
 
 #endif /* ENABLE_SOUND */
-- 
2.37.3

